config/validation.go

```diff
@@ -0,0 +1,86 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+
+	validation "github.com/go-ozzo/ozzo-validation/v4"
+)
+
+// Validate validate the config as an input. If not valid, it returns error
+func Validate(conf Config) error {
```

Comment:
Do we need a single validate for both `ClientConfig` and `ServerConfig`.

Reply:
to validate either `ClientConfig` or `ServerConfig` we can simply call single function, `config.Validate()`. I don't have a strong opinion about it. If splitting the validate function into 2 is better, we can do it